runtime/lua/vim/lsp/handlers.lua

```diff
@@ -16,68 +16,76 @@ local function err_message(...)
   api.nvim_command('redraw')
 end
 
---see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_executeCommand
+-- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_executeCommand
+---@type vim.lsp.ResponseHandler
 M[ms.workspace_executeCommand] = function(_, _, _, _)
   -- Error handling is done implicitly by wrapping all handlers; see end of this file
 end
 
---see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
----@param result lsp.ProgressParams
----@param ctx lsp.HandlerContext
-M[ms.dollar_progress] = function(_, result, ctx)
+-- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
+---@param params lsp.ProgressParams
+---@type vim.lsp.NotificationHandler
+M[ms.dollar_progress] = function(_, params, ctx)
   local client = vim.lsp.get_client_by_id(ctx.client_id)
   if not client then
     err_message('LSP[id=', tostring(ctx.client_id), '] client has shut down during progress update')
+    -- TODO(wookayin): This is a notification handler so should be removed (after testing!) #16472
```

Comment:
Without `vim.NIL` something bad will happen (#16472). It looks like some kind of ACK is sent to LSP, so now I realize that a notification handler apparently needs to return `vim.NIL` to indicate a success. (Returning `nil` will be like an error, which is more consistent with the behavior of request handlers).